watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +375,288 @@ def _interp_nan_values(global_values, global_results):
 
 # ================================================================
 
-def parameter_sweep(model, sweep_params, outputs, results_file=None, optimize_function=_default_optimize,
-        optimize_kwargs=None, reinitialize_function=None, reinitialize_kwargs=None,
-        mpi_comm=None, debugging_data_dir=None, interpolate_nan_outputs=False, num_samples=None, seed=None):
+def _create_local_output_skeleton(model, sweep_params, num_samples):
 
-    '''
-    This function offers a general way to perform repeated optimizations
-    of a model for the purposes of exploring a parameter space while
-    monitoring multiple outputs. 
-    If provided, writes single CSV file to ``results_file`` with all inputs and resulting outputs.
+    output_dict = {}
+    output_dict["sweep_params"] = {}
+    output_dict["outputs"] = {}
 
-    Arguments:
+    # 1. deal with the inputs
+    for key in sweep_params.keys():
+        var = sweep_params[key].pyomo_object
+        var_str = sweep_params[key].pyomo_object.name
+        output_dict["sweep_params"][var_str] =  _create_component_output_skeleton(var, num_samples)
 
-        model : A Pyomo ConcreteModel containing a watertap flowsheet, for best 
-                results it should be initialized before being passed to this
-                function.
+    # 2. lets deal with the output variables from the pyomo model
+    for var in variables_in_activated_equalities_set(model.fs):
+        var_str = var.name
+        output_dict["outputs"][var_str] = _create_component_output_skeleton(var, num_samples)
 
-        sweep_params: A dictionary containing the values to vary with the format
-                      ``sweep_params['Short/Pretty-print Name'] =
-                      (model.fs.variable_or_param[index], lower_limit, upper_limit, num_samples)``.
-                      A uniform number of samples ``num_samples`` will be take between
-                      the ``lower_limit`` and ``upper_limit``.
+    # 3. lets deal with expressions
+    for expr in expressions_set(model.fs):
+        expr_str = expr.name
+        output_dict["outputs"][expr_str] = _create_component_output_skeleton(var, num_samples)
 
-        outputs : A dictionary containing "short names" as keys and and Pyomo objects
-                  on ``model`` whose values to report as values. E.g.,
-                  ``outputs['Short/Pretty-print Name'] = model.fs.variable_or_expression_to_report``.
+    # 4. Lets deal with objectives
+    for obj in total_objectives_set(model):
+        obj_str = obj.name
+        output_dict["outputs"][obj_str] = _create_component_output_skeleton(var, num_samples)
```

Comment:
The three `for` loops above could be replaced with
```
for pyo_obj in model.component_data_objects((Var, Expression, Objective), active=True):
   output_dict["outputs"][pyo_obj.name] = _create_component_output_skeleton(pyo_obj, num_samples)
```

Reply:
incorporated.